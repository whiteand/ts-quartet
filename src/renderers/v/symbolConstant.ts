import { ISchemaRenderer } from "../../types";
import { getStrictComparisonWithSchemaRenderer } from "./utils";

export const symbolConstantRenderer: ISchemaRenderer<
  symbol
> = getStrictComparisonWithSchemaRenderer("validSymbol");
