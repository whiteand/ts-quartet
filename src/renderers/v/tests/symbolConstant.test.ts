import { symbolConstantRenderer } from "../symbolConstant";
import { snapshot } from "./snapshot";

describe("v constant number renderer", () => {
  test("v(Symbol.for('quartet'))", () => {
    snapshot(symbolConstantRenderer, { schema: Symbol.for("Andrew") });
  });
});
