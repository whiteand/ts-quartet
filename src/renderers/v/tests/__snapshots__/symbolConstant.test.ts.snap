// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v constant number renderer v(Symbol.for('quartet')) 1`] = `"_valueId === _validSymbol_undefined"`;

exports[`v constant number renderer v(Symbol.for('quartet')) 2`] = `"_valueId !== _validSymbol_undefined"`;

exports[`v constant number renderer v(Symbol.for('quartet')) 3`] = `"if (_valueId === _validSymbol_undefined) return true;"`;

exports[`v constant number renderer v(Symbol.for('quartet')) 4`] = `"if (_valueId !== _validSymbol_undefined) return false;"`;
